internal/app/machined/pkg/system/services/etcd.go

```diff
@@ -574,8 +568,8 @@ func (e *Etcd) argsForControlPlane(ctx context.Context, r runtime.Runtime) error
 		"listen-peer-urls":                   "https://" + net.FormatAddress(listenAddress) + ":2380",
 		"listen-client-urls":                 "https://" + net.FormatAddress(listenAddress) + ":2379",
 		"client-cert-auth":                   "true",
-		"cert-file":                          constants.KubernetesEtcdPeerCert,
```

Comment:
why were we using the peer cert instead of the actual cert here?

Reply:
it was a bug, but in fact peer and server certs were generated with same settings, so nobody noticed.. we have same CA for server and peer connections, so they just work